pkg/agent/flowexporter/exporter.go

```diff
@@ -353,11 +395,57 @@ func (exp *FlowExporter) fillEgressInfo(conn *connection.Connection) {
 	}
 }
 
+// getServiceName returns the name of the service from the set of services provided
+// which has the matching provided port and the name of the port.
+// Empty strings are returned if no service match is found.
+func getServiceName(port uint16, services []*corev1.Service) (string, string) {
+	for _, service := range services {
+		for _, servicePort := range service.Spec.Ports {
+			if servicePort.Port == int32(port) {
+				return service.Name, servicePort.Name
+			}
+		}
+	}
+
+	return "", ""
+}
+
+// FillServiceInfo updates the given conn of type FlowTypeToExternal with the name of
+// the service whos port matches the destination port. An error is returned if no match
+// is found or errors occurred retrieving services
+func (exp *FlowExporter) FillServiceInfo(conn *connection.Connection) error {
+	errorString := "Failed to fill serviceInfo for connection"
+	if exp.serviceInformer == nil {
+		return fmt.Errorf("%s: serviceInformer is nil", errorString)
+	}
+	serviceLister := exp.serviceInformer.Lister()
+	if serviceLister == nil {
+		return fmt.Errorf("%s: serviceLister is nil", errorString)
+	}
+	serviceNamespaceLister := serviceLister.Services(conn.DestinationPodNamespace)
+	if serviceNamespaceLister == nil {
+		return fmt.Errorf("%s: serviceNamespaceLister is nil", errorString)
+	}
+	services, err := serviceNamespaceLister.List(labels.NewSelector())
+	if err != nil {
+		return fmt.Errorf("%s: serviceNamespaceLister failed to List: %w", errorString, err)
+	}
+
+	matchingServiceName, portName := getServiceName(conn.FlowKey.DestinationPort, services)
```

Comment:
`getServiceName` matches on the service's `port` field which is used within the cluster whereas the `nodePort` field would be the one that matches the dest port of the external ingress traffic.